chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -936,6 +936,29 @@ impl PeerManagerActor {
             .collect::<Vec<_>>()
     }
 
+    // Get peers that are potentially unreliable and we should avoid routing messages through them.
+    // Currently we're picking the peers that are NOT the highest peers.
+    fn unreliable_peers(&self) -> HashSet<PeerId> {
+        let connected_peers = self.state.connected_peers.read();
+        // This finds max height among peers, and returns one peer close to such height.
+        let max_height = match (connected_peers.values())
+            .map(|connected_peer| connected_peer.full_peer_info.chain_info.height)
+            .max()
+        {
+            Some(height) => height,
+            None => return HashSet::default(),
+        };
```

Comment:
Is this abusable? What if a peer lies to us that his max height is big, making everyone else unreliable? As far as I see, we don't try to verify height elsewhere, we just seem to believe whatevere the peer tells to us

Reply:
We actually do verify the height that we get from the peer (by asking it to send us the BlockHeaders if the height it too large) - but that does come with a delay.

Reply:
Discussed offline: we can reduce trust here if we just look at the peers which are not far behind us.